services/supervisor/src/lib.rs

```diff
@@ -299,11 +300,13 @@ impl Service for Supervisor {
     }
 
     fn after_transactions(&self, mut context: CallContext<'_>) -> Result<(), ExecutionError> {
+        skip_for_genesis!(context);
+
         let mut schema = Schema::new(context.service_data());
         let configuration = schema.supervisor_config();
         let core_schema = context.data().for_core();
-        let validator_count = core_schema.consensus_config().validator_keys.len();
         let height = core_schema.height();
```

Comment:
It seems that `height` is used only as `height.next()` further down in the code. Thus, could it make sense to assign `let height = core_schema.next_height()` and *not* skip the genesis block?